### PR TITLE
Feature/di container

### PR DESCRIPTION
**Pull request by iruzevic**
[x] Moving all Eightshift-project stuff to separate folders and creating a CLI for each of them.

TODO:
1. Create option to run from WPCLI and not from php script:
[x] setup
[ ] import
[ ] export
[ ] deploy 

2. Refactor lint.php to work the same way
3. Create combined CLI command for all Eightshift-project stuff

### CODE REVIEW

File: src/setup/Setup.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      \WP_CLI::warning( 'No core version is defined. Skipping.' );
+    }
+  }
+
+  // Check if plugins key exists in config.
+  if ( ! $skip_plugins ) {
+
+  $plugins = $data['plugins'] ?? [];
+
+  if ( ! empty( $plugins ) ) {
```
Comment (IvanGrginovInf) [inline src/setup/Setup.php:62]: Indentation or something else seems to be wrong here.

File: src/setup/Setup.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        foreach( $plugins_github as $name => $version ) {
+          \WP_CLI::runcommand( "plugin install https://github.com/{$name}/archive/{$version}.zip --force" );
+          \WP_CLI::log( '--------------------------------------------------' );
+        }
+      } else {
+        \WP_CLI::warning( 'No Github plugins are defined. Skipping.' );
+      }
+    }
+  }
+  }
```
Comment (IvanGrginovInf) [inline src/setup/Setup.php:96]: ⬆️

**Review by IvanGrginovInf — CHANGES_REQUESTED** [review]
Left a few comments, otherwise looks good!

File: src/setup/UpdateCli.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  public function __invoke( array $args, array $assoc_args ) {
+
+    // Get Props.
+    $root = $assoc_args['root'] ?? static::OUTPUT_DIR;
+
+    // Read the template contents, and replace the placeholders with provided variables.
+    $class = $this->get_example_template( __DIR__, $this->get_class_short_name() );
+
+    // $reflection_class = new \ReflectionClass( $item );
+    // $class            = $reflection_class->newInstanceArgs( [ null ] );
```
Comment (IvanGrginovInf) [inline src/setup/UpdateCli.php:72]: Commented out code

**Review by IvanGrginovInf — APPROVED** [review]
Left some commented code but otherwise awesome!

**Review by IvanGrginovInf — APPROVED** [review]

**Review by gabriel-glo — APPROVED** [review]

**Pull request merged**